lib/declarative_authorization/authorization.rb

```diff
@@ -230,6 +231,9 @@ def permit!(privilege, options = {})
           return true if authorized
         else
           puts "Rule has #{rule.attributes.count} attributes, examining them:"
+    
+          all_attributes_matched = true
```

Comment:
Should we invert this and default it to false? It feels more intuitive and safe to start both of these values as false and only consider them true once the conditions have been met.

Reply:
Makes sense. 

I updated the logic to track [matching_attributes_count](https://github.com/appfolio/ae_declarative_authorization/blob/777b4b9c3812f4305d928fcfa5136d1d45210537/lib/declarative_authorization/authorization.rb#L235C11-L235C36) and [matching_conditions_count](https://github.com/appfolio/ae_declarative_authorization/blob/777b4b9c3812f4305d928fcfa5136d1d45210537/lib/declarative_authorization/authorization.rb#L251). We can use those two values to determine if the individual conditions in each attribute are met, as well as if all/any of the attributes are satisfied (depending on if the rule uses the `and` or `or` operators)